zcash_client_sqlite/src/wallet.rs

```diff
@@ -2020,9 +2022,29 @@ pub(crate) fn truncate_to_height<P: consensus::Parameters>(
         named_params![":new_end_height": u32::from(block_height + 1)],
     )?;
 
-    // If we're removing scanned blocks, we need to truncate the note commitment tree, un-mine
-    // transactions, and remove received transparent outputs and affected block records from the
-    // database.
+    // Mark transparent utxos as un-mined. Since the TXO is now not mined, it would ideally be
+    // considered to have been returned to the mempool; it _might_ be spendable in this state, but
+    // we must also set its max_observed_unspent_height field to NULL because the transaction may
+    // be rendered entirely invalid by a reorg that alters anchor(s) used in constructing shielded
+    // spends in the transaction.
+    conn.execute(
+        "UPDATE transparent_received_outputs
+         SET max_observed_unspent_height = NULL
+         WHERE max_observed_unspent_height > :height",
+        named_params![":height": u32::from(block_height)],
+    )?;
```

Comment:
The output might be invalid at the truncation height; i.e. it might have been created in a transaction where the anchor used to spend notes in order to create the output is not valid. So we can't say anything about the max observed unspent height; we have to re-observe the output in order to set it.

Reply:
By construction, `tx.anchor_height < tx.mined_height`. Ergo, per my comment, if `tx.mined_height <= :height` then `tx.anchor_height < :height` and thus the anchor is still valid.